python/ray/data/datasource/parquet_datasource.py

```diff
@@ -56,6 +56,61 @@ def _deregister_parquet_file_fragment_serialization():
     ray.util.deregister_serializer(ParquetFileFragment)
 
 
+# this retry helps when HA hdfs service not able to handle overloaded read request.
+# even with HA configed, hdfs service might be overloaded with parquet file read
+# request expecially when simutaneously running many hyper parameter tuning jobs
+# with ray.data parallelism setting at high value like the default 200
+def _deserialize_pieces(
+    serialized_pieces: str,
+) -> List["pyarrow._dataset.ParquetFileFragment"]:
+    from ray import cloudpickle
+
+    min_interval = 0
+    final_exception = None
+    # retry at most 8 times
+    for i in range(8):
+        try:
+            pieces: List["pyarrow._dataset.ParquetFileFragment"] = cloudpickle.loads(
+                serialized_pieces
+            )
+            return pieces
+        except Exception as e:
```

Comment:
Can we retry for just connection error?

Reply:
I think we should retry for any error, since as far as the data reading can be recovered from retry, why not do this for any type of error?  For example, a network glitch causing data reading timeout or something.

Reply:
Not all errors are transient, e.g. bad request, auth error.